src/main/java/group1/mips_simulator/SimController.java

```diff
@@ -69,41 +71,144 @@ protected void restartComputer() {
     public TextField PC;
     public TextField MAR;
     public TextField MBR;
-    public TextField RI;
+    public TextField IR;
     public TextField MFR;
     public TextField Privileged;
     public TextField CC;
 
     public Button PC_button;
     public Button MAR_button;
     public Button MBR_button;
-    public Button RI_button;
+    public Button IR_button;
 
     //endregion
 
     //endregion
 
     //region Command Buttons
 
-    public void loadOnClick() {
+    @FXML
+    public Button loadButton;
+    @FXML
+    public Button loadPlusButton;
+    @FXML
+    public Button storeButton;
+    @FXML
+    public Button storePlusButton;
+
+    @FXML
+    public Button run_Button;
+    @FXML
+    public Button step_Button;
+    @FXML
+    public Button halt_Button;
+
+    /**
+     * Use the value in the Memory Address Register, go to that location in Memory
+     * and read the value from memory into the Memory Buffer Register
+     */
+    public void loadOnClick(ActionEvent actionEvent) {
+        Register memAddReg = this.computer.cpu.regfile.getMAR();
+        short valueInMemory = this.computer.mem.read(memAddReg.read());
+        this.computer.cpu.regfile.getMBR().write(valueInMemory);
+        redraw();
     }
 
+    /**
+     * Similar to the loadOnClick function (read value from memory pointed
+     * to by MAR into MBR) but this will also increase the MAR AFTER
+     * the read.
+     */
     public void loadPlusOnClick(ActionEvent actionEvent) {
+        loadOnClick(actionEvent);
+        Register memAddrReg = this.computer.cpu.regfile.getMAR();
```

Comment:
minor adjustment, non-blocker
consistent naming of memAddrReg vs memAddReg

personally like memAddrReg, but memAddReg fine too and requires fewer modifications

Reply:
Just so I'm clear, you're asking that I remove the `r` small-r character from `memAddrReg`? Will update.

Reply:
Also, a pro tip for GitHub. When you leave a comment there's a button that looks like a page with a + - character on it called "Suggest a Change" 
![image](https://github.com/akm303/MIPS_Simulator/assets/25259855/932e3992-4310-4b50-a2e5-0ebfaba7049f)

If you click that, you can make a change to a single line and make a cool interface to allow the merge author to make the change with a single button click